File: Sources/Stopwatch/Periods.swift
import Foundation

enum Period: String, CaseIterable {
    case morning
    case afternoon
    case night

    var label: String {
        switch self {
        case .morning: return "Morning"
        case .afternoon: return "Afternoon"
        case .night: return "Night"
        }
    }

    static func current(at date: Date = Date()) -> Period {
        let hour = Calendar.current.component(.hour, from: date)
        switch hour {
        case 5..<12: return .morning
        case 12..<18: return .afternoon
        default: return .night
        }
    }

    func contains(hour: Int) -> Bool {
        switch self {
        case .morning: return (5...11).contains(hour)
        case .afternoon: return (12...17).contains(hour)
        case .night: return hour >= 18 || hour < 5
        }
    }
}

File: linux/stopwatch_linux/periods.py
from enum import Enum


class Period(Enum):
    MORNING = "morning"
    AFTERNOON = "afternoon"
    NIGHT = "night"

    @property
    def label(self) -> str:
        return {
            Period.MORNING: "Morning",
            Period.AFTERNOON: "Afternoon",
            Period.NIGHT: "Night",
        }[self]

    def contains_hour(self, hour: int) -> bool:
        if self is Period.MORNING:
            return 5 <= hour <= 11
        if self is Period.AFTERNOON:
            return 12 <= hour <= 17
        return hour >= 18 or hour < 5

    @classmethod
    def ordered(cls):
        return [cls.MORNING, cls.AFTERNOON, cls.NIGHT]
